update_checker.go

```diff
@@ -0,0 +1,44 @@
+//go:build !docker
+
+package main
+
+import (
+	"context"
+	"time"
+
+	"github.com/suborbital/subo/subo/release"
+	"github.com/suborbital/subo/subo/util"
+)
+
+const checkVersionTimeout = 500 * time.Millisecond
+
+func checkForUpdates() {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+	versionChan := make(chan string)
+
+	go func() {
+		msg := ""
+		if version, err := release.CheckForLatestVersion(); err != nil {
+			msg = err.Error()
+		} else if version != "" {
+			msg = version
+		}
+
+		select {
+		case <-ctx.Done():
```

Comment:
this is confusing here. The only time this is ran is when some time has elapsed and the time.after triggers, the function gets to the end, the deferred cancel is called, and the code already returned from release.checkforlatestversion.

Alternatively what I would do is create a context.withtimeout, pass that one to the checkforlatestversion, propagate it across until github.NewClient.Repositories.GetLatestRelease, because the first argument there is a context.

That way the github sdk will return the "context deadline exceeded" error, and we don't have to deal with that here.